packages/@uppy/core/src/index.js

```diff
@@ -733,8 +734,12 @@ class Uppy {
 
     this.setState({ files })
 
-    newFiles.forEach((newFile) => {
+    newFiles.forEach((newFile, i) => {
       this.emit('file-added', newFile)
+
+      if (i + 1 === newFiles.length) {
+        this.emit('final-file-added', newFile)
+      }
```

Comment:
I think this event looks geared towards a very particular use case, and could be more general. I'd expect it's pretty rare that a developer actually wants to receive only the last file in a particular set that the user added at once. We could instead give them access to the whole list of newly added files. Perhaps like this:

```js
this.emit('files-added', newFiles)
```

This still allows you to do the same thing as final-file-added:
```js
uppy.on('files-added', (list) => {
  const last = list[list.length - 1]
})
```
but also _more_ things.